### PR TITLE
Fixes #25397 - reimplement indent macro

### PR DESCRIPTION
**Pull request by lzap**
It was leaving some garbage whitespace on the last indented line.

### CODE REVIEW

File: lib/foreman/renderer/scope/macros/base.rb [hunk 1]
```diff
@@ -46,10 +46,20 @@ def save_to_file(filename, content)
...
 
-          def indent(count)
+          def indent(count, skip1: false)
             return unless block_given? && (text = yield.to_s)
-            prefix = " " * count
-            prefix + text.gsub(/\n/, "\n#{prefix}")
+            prefix = ' ' * count
+            result = []
+            line_no = 0
+            text.each_line do |line|
```
Comment (stbenjam) [inline lib/foreman/renderer/scope/macros/base.rb]: Why not `with_index`?

```ruby
text.each_line.with_index do |line, line_no|
  # ...
end
```

File: lib/foreman/renderer/scope/macros/base.rb [hunk 2]
```diff
@@ -46,10 +46,20 @@ def save_to_file(filename, content)
...
+              if line_no != 0 || !skip1
```
Comment (stbenjam) [inline lib/foreman/renderer/scope/macros/base.rb]: Flipping things around seems clearer to me

```ruby
if line_no == 0 && skip1
  result << line
else
...
```

**Comment by stbenjam** [comment]
Otherwise :+1:

File: lib/foreman/renderer/scope/macros/base.rb [hunk 3]
```diff
@@ -46,10 +46,20 @@ def save_to_file(filename, content)
             "cat << EOF > #{filename}\n#{content}EOF"
           end
 
-          def indent(count)
+          def indent(count, skip1: false)
```
Comment (iNecas) [inline lib/foreman/renderer/scope/macros/base.rb:49]: Probably `skip_first_line` would be more descriptive, than `skip1`.

Comment (lzap) [reply lib/foreman/renderer/scope/macros/base.rb:49]: Yeah that was my initial idea but feels quite long. If there is one more objection I will rebase ;-)

File: lib/foreman/renderer/scope/macros/base.rb (hunk 1 above)
Comment (lzap) [reply lib/foreman/renderer/scope/macros/base.rb]: Oh nice, I could not figure out this one :-)

File: lib/foreman/renderer/scope/macros/base.rb (hunk 2 above)
Comment (lzap) [reply lib/foreman/renderer/scope/macros/base.rb]: Sure, happens when you refactor your code...

**Comment by lzap** [comment]
Pushed.

**Comment by lzap** [comment]
Tests are green!

**Comment by lzap** [comment]
Damn guys we left this one to rot here but we merged template change, now develop is broken. Let's merge this one? @iNecas @mmoll Thanks.

File: lib/foreman/renderer/scope/macros/base.rb (hunk 3 above)
Comment (iNecas) [reply lib/foreman/renderer/scope/macros/base.rb:49]: Ok, since it's already used in templates, I will not insist anymore

**Review by iNecas — APPROVED** [review]

**Comment by iNecas** [comment]
@lzap please rebase just ot make sure the CI haven't run red since the last time it run

**Comment by tbrisker** [comment]
Merging now to unblock branching, this was missed long time ago.

**Pull request merged**